### PR TITLE
[FEATURE] Exclude some html parts to Typo3PageContentExtractor with …

### PR DESCRIPTION
**Pull request by kalypso63**
[FEATURES] Exclude some html parts to Typo3PageContentExtractor with classname & configure it with typoscript (plugin.tx_solr.index.queue.pages.excludeHtmlPartWithClassname)

### CODE REVIEW

File: Classes/Typo3PageContentExtractor.php [hunk 1]
```diff
@@ -59,6 +59,24 @@ protected function extractContentMarkedForIndexing($html)
             $html, $indexableContents);
         $indexableContent = implode($indexableContents[0], '');
 
+        //exclude some html part with classname (.exclude-solr-part)
```
Comment (irnnr) [inline Classes/Typo3PageContentExtractor.php]: Could you please move this to a separate method (and ideally adda unit test).

File: Configuration/TypoScript/Solr/setup.txt [hunk 2]
```diff
@@ -97,6 +97,9 @@ plugin.tx_solr {
 				// Only index standard pages and mount points that are not overlayed.
 				additionalWhereClause = (doktype = 1 OR (doktype=7 AND mount_pid_ol=0)) AND no_search = 0
 
+                //exclude some html parts inside TYPO3SEARCH markers with classname (comma list)
```
Comment (irnnr) [inline Configuration/TypoScript/Solr/setup.txt]: I'd rather name this `excludeContentByClass` and use a default `typo3-search-exclude`

**Comment by timoschmidt** [comment]
Nice improvement and also thx for adding a unit test!

**Comment by irnnr** [comment]
:+1:

**Pull request merged**